x-pack/legacy/plugins/reporting/server/routes/generation.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import boom from 'boom';
+import { API_BASE_URL } from '../../common/constants';
+import {
+  ServerFacade,
+  ExportTypesRegistry,
+  HeadlessChromiumDriverFactory,
+  RequestFacade,
+  ReportingResponseToolkit,
+  Logger,
+} from '../../types';
+import { registerGenerateFromJobParams } from './generate_from_jobparams';
+import { registerGenerateCsvFromSavedObject } from './generate_from_savedobject';
+import { registerGenerateCsvFromSavedObjectImmediate } from './generate_from_savedobject_immediate';
+import { registerLegacy } from './legacy';
+import { createQueueFactory, enqueueJobFactory } from '../lib';
+
+export function registerJobGenerationRoutes(
+  server: ServerFacade,
+  exportTypesRegistry: ExportTypesRegistry,
+  browserDriverFactory: HeadlessChromiumDriverFactory,
+  logger: Logger
+) {
+  const config = server.config();
+  const DOWNLOAD_BASE_URL = config.get('server.basePath') + `${API_BASE_URL}/jobs/download`;
```

Comment:
You could access this from NP with `core.http.basePath.serverBasePath` instead of relying on the legacy config.